### PR TITLE
all in one simple complex example

### PR DESCRIPTION
**Pull request by marinaluisa2020**
simple command, multi function, tested and running 28/02/2020

SERVER ESTABLISH, 10 YEARS ONLINE, SIMPLE-API-RESTFUL-PROXY-RANDOM 20 IP PER IP PER DAY (USE AVG SECURE VPN FOR MORE IP)
28/02/2020 - 22:08: TESTED AND RUNNING FOR 1.5.2--><--1.12.2 SERVERS, ONLY TYPE WINDOWS CMD AND RUN

I stopped here, I hope this is useful for someone as it would have been useful for me.

use simple windows notepad for edit, it's faster and more practical and works only with your mind and creativity

### CODE REVIEW

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 1]
```diff
@@ -0,0 +1,230 @@
+/**
+
+/npm install mineflayer
+/npm install socks
+/npm install proxy-agent
+/npm install https
+
+/node chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: This name is too long

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 2]
```diff
@@ -0,0 +1,230 @@
+/**
+
+/npm install mineflayer
+/npm install socks
+/npm install proxy-agent
+/npm install https
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: What are those slashes meant for?

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+/node chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt
+
+SERVER ESTABLISH, 10 YEARS ONLINE, SIMPLE-API-RESTFUL-PROXY-RANDOM 20 IP PER IP PER DAY (USE AVG SECURE VPN FOR MORE IP)
+28/02/2020 - 22:08: TESTED AND RUNNING FOR 1.5.2--><--1.12.2 SERVERS, ONLY TYPE WINDOWS CMD AND RUN
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: This message is not descriptive. Dunno what's going on in there.

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+*/
+
+const mc = require('mineflayer')
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Should be `mineflayer` as in other examples.

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+const socks = require('socks').SocksClient
+const ProxyAgent = require('proxy-agent')
+const https = require('https');
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: This file hasn't been linted with standard

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+
+var cco = 0;
+const tta = process.argv[2]
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: What's the meaning of those variables?

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+  console.log("#############ERRO#####################");
+  console.log(e);
+  console.log("#############ERRO#####################");
+setTimeout(trystart(botname, target), 9000);
+}
+finally {
+  console.log("entering and leaving the finally block");
+}
+},9000);
+}
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: 0 indent at all

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+
+start(botname, target)
+
+//throw "thrown message";
+}
+catch (e) 
+{
+  console.log("#############ERRO#####################");
+  console.log(e);
+  console.log("#############ERRO#####################");
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Flood and typo

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+setTimeout(trystart(botname, target), 9000);
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Memory leak

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+}
+catch (e) 
+{
+  console.log("#############ERRO#####################");
+  console.log(e);
+  console.log("#############ERRO#####################");
+setTimeout(tryloop(i), 9000);
+}
+finally {
+  console.log("entering and leaving the finally block");
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Unneeded

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+}
+catch (e) 
+{
+  console.log("#############ERRO#####################");
+  console.log(e);
+  console.log("#############ERRO#####################");
+setTimeout(trystart(botname, target), 9000);
+}
+finally {
+  console.log("entering and leaving the finally block");
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Unneeded

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+  // The whole response has been received. Print out the result.
+  resp.on('end', () => {
+    console.log(JSON.parse(data).explanation);
+console.log(data);
+console.log(JSON.parse(data).ip);
+
+
+//-------->START
+const proxyHost = JSON.parse(data).ip;
+const proxyPort = JSON.parse(data).port;
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Save it to a variable, JSON.parse is inefficient

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+  connect: client => {
+    socks.createConnection({
+      proxy: {
+        host: proxyHost,
+        port: proxyPort,
+        type: 5
+      },
+      command: 'connect',
+      destination: {
+        host: "legacy.craftlandia.com.br",
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: ip should be as a program argument

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+      if (err) {
+        console.log("########################err###############")
+        trystart(botname, target);
+        console.log(err)
+        console.log(err)
+        console.log(err)
+        console.log(err)
+        console.log(err)
+        console.log(err)
+        console.log("########################err###############")
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: May be already handled by tryloop/trystart. Also span and flood

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+        return
+      }
+
+      client.setSocket(info.socket)
+      client.emit('connect')
+    })
+  },
+  agent: new ProxyAgent({ protocol: 'socks5', host: proxyHost, port: proxyPort }),
+  username: botname,
+  version: "1.12.2"
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Version as a param

File: examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+
+bot.on('kicked', (m1,m2) => {
+  console.log(m1)
+console.log(m1)
+console.log(m1)
+console.log(m1)
+console.log(m1)
+console.log(m1)
+console.log(m1)
+console.log(m1)
```
Comment (wvffle) [inline examples/chat.msg.log.debug.proxy.time.schedule.loop.random.proxy-api-restful.js.txt]: Spam

**Comment by wvffle** [comment]
Also judging by the:
- code style
- server address in the example
- nick
- join date
You're @Escanor2019, whom we banned for being rude towards me.

**Comment by IdanHo** [comment]
i agree, and even ignoring the horrendous code quality, this example does not add anything informative to our current examples lineup, if users of mineflayer cant combine the javascript from multiple files themselves they should learn javascript first.

**Comment by marinaluisa2020** [comment]
only poor people try to learn, entrepreneurs try to execute.

**Pull request closed without merging**